### PR TITLE
Add license script instructions

### PR DESCRIPTION
**Pull request by amzn-changml**
## Change summary

Adds instructions for scanning licenses in the codebase for generating a NOTICES file and a SPDX tag summarization file during the project development process

### Submission Checklist:

* [x] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/build/distributable-engine.md [hunk 1]
```diff
@@ -224,3 +224,21 @@ Engine developers will work on O3DE source code and create new distributable bui
 Project developers will work on project-specific source code or use O3DE creative tools to add project content. To enable their workflows, project developers are required to _only_ pull a project from its hosted location.
 
 Whenever working on their project, project developers should launch the tools that were checked in as the distibutable engine build. These tools are located in the project's `install\bin\Windows\profile\Default` subdirectory.
+
+### License file generation
+
+License attribution files can be generated during the project development process to properly attribute any code or packages that were imported (often called the NOTICES file). To scan project directories for licenses, project developers can run a script located in the engine's `scripts\license_scanner` subdirectory.
+
+In this example, with the engine, project, and `C:\o3de-source\3rdParty` as your 3rd Party packages folder, run the following script in the engine source folder
```
Comment (willihay) [inline content/docs/user-guide/build/distributable-engine.md:232]:
```suggestion
In this example, with the engine, project, and `C:\o3de-source\3rdParty` as your 3rd Party packages folder, run the following script in the engine source folder:
```

File: content/docs/user-guide/build/distributable-engine.md [hunk 2]
```diff
@@ -224,3 +224,21 @@ Engine developers will work on O3DE source code and create new distributable bui
...
+   ```cmd
+   python\python.cmd scripts\license_scanner\license_scanner.py ^
+     --config-file scripts\license_scanner\scanner_config.json ^
+     --license-file-path build\windows_vs2019\NOTICES.txt ^
+     --package-file-path build\windows_vs2019\spdx-packages.json ^
+     --scan-path C:\o3de-source W:\MyProject C:\o3de-source\3rdParty
+   ```
+This will scan the engine, project, and `C:\o3de-source\3rdParty` folders for license files using a default configuration in the `scanner_config.json` file, then generate a `NOTICES.txt` in the build output folder. In addition, if a `PackageInfo.json` file is detected in `C:\o3de-source\3rdParty` for each package, the script will add its package license manifest to a summerized file called `spdx-packages.json`.
```
Comment (willihay) [inline content/docs/user-guide/build/distributable-engine.md:240]: Suggest rewording the last sentence. Did I keep the meaning intact?
```suggestion
This will scan the engine, project, and `C:\o3de-source\3rdParty` folders for license files using a default configuration in the `scanner_config.json` file, then generate a `NOTICES.txt` in the build output folder. In addition, if a `PackageInfo.json` file is detected in `C:\o3de-source\3rdParty`, the script will scan the file and add a package license manifest for each package to a summary file called `spdx-packages.json`.
```

File: content/docs/user-guide/build/distributable-engine.md [hunk 3]
```diff
@@ -224,3 +224,21 @@ Engine developers will work on O3DE source code and create new distributable bui
...
+
+{{< note >}}
+The scanner script will perform filename scans based on the configuration of the `scanner_config.json` file, but it is not guarenteed to find all license files, nor catch recurisve dependancies. To ensure there is sufficent attribution in your project, we recommend tools such as Scancode Toolkit: https://github.com/nexB/scancode-toolkit and Fossology https://www.fossology.org/ to further validate your dependancies.
```
Comment (willihay) [inline content/docs/user-guide/build/distributable-engine.md:243]:
```suggestion
The scanner script will perform filename scans based on the configuration of the `scanner_config.json` file, but it is not guaranteed to find all license files, nor catch recursive dependencies. To ensure there is sufficient attribution in your project, we recommend tools such as [Scancode Toolkit](https://github.com/nexB/scancode-toolkit) and [Fossology](https://www.fossology.org) to further validate your dependencies.
```

File: content/docs/user-guide/build/distributable-engine.md [hunk 4]
```diff
@@ -224,3 +224,21 @@ Engine developers will work on O3DE source code and create new distributable bui
 Project developers will work on project-specific source code or use O3DE creative tools to add project content. To enable their workflows, project developers are required to _only_ pull a project from its hosted location.
 
 Whenever working on their project, project developers should launch the tools that were checked in as the distibutable engine build. These tools are located in the project's `install\bin\Windows\profile\Default` subdirectory.
+
+### License file generation
```
Comment (willihay) [inline content/docs/user-guide/build/distributable-engine.md:228]: Assuming this license file may need to go into a project's release layout, I'm wondering if you want to add a section and paragraph to the packaging [Release Layout tutorial](https://www.o3de.org/docs/user-guide/packaging/windows-release-builds/) that links to this topic.

**Review by willihay — CHANGES_REQUESTED** [review]

File: content/docs/user-guide/build/distributable-engine.md (hunk 2 above)
Comment (amzn-changml) [reply content/docs/user-guide/build/distributable-engine.md:240]: Yup seems reasonable

File: content/docs/user-guide/build/distributable-engine.md (hunk 3 above)
Comment (amzn-changml) [reply content/docs/user-guide/build/distributable-engine.md:243]: That was some bad spelling on my part. The suggestions look good.

File: content/docs/user-guide/build/distributable-engine.md (hunk 4 above)
Comment (amzn-changml) [reply content/docs/user-guide/build/distributable-engine.md:228]: Sure, that makes sense. I should probably add some additional info in [O3DE Packages](https://www.o3de.org/docs/user-guide/build/packages/) as well.

Comment (willihay) [reply content/docs/user-guide/build/distributable-engine.md:228]: For clarity, since I just re-read what I said and didn't get the same meaning out of it twice, I meant to say "add a new paragraph" AND THEN "in that new paragraph, add a link". Not sure if that is how it came across to you. lol

Comment (amzn-changml) [reply content/docs/user-guide/build/distributable-engine.md:228]: It came across as you intended, but thanks for the clarification 😉

**Comment by willihay** [comment]
Adding a do-not-merge so that author can add additional updates.

File: content/docs/user-guide/build/distributable-engine.md (hunk 4 above)
Comment (amzn-changml) [reply content/docs/user-guide/build/distributable-engine.md:228]: Added paragraphs and links to each of the sections discussed

File: content/docs/user-guide/packaging/windows-release-builds.md [hunk 5]
```diff
@@ -260,6 +260,10 @@ Follow the instructions in [Bundling Assets for a Project Game Release Layout](a
 
 You may notice that both your bundled assets are smaller than the default `engine.pak` that was created automatically when you created a project game release layout earlier. 
 
+## License file generation
+License attribution files can be generated during the project development process to properly attribute any code or packages that were imported (often called the NOTICES file). To scan project directories for licenses, you can run a script located in the engine's scripts\license_scanner subdirectory.
+
+Refer to the instructions in [Distributable Engine](content/docs/user-guide/build/distributable-engine.md#license-file-generation) for more details. Generally, you will copy the generated file to the root of your game release layout to properly attribute your dependencies.
```
Comment (willihay) [inline content/docs/user-guide/packaging/windows-release-builds.md:266]: Fix broken link and link text.
```suggestion
Refer to the instructions in [Engine and Project Distribution](/docs/user-guide/build/distributable-engine#license-file-generation) for more details. Generally, you will copy the generated file to the root of your game release layout to properly attribute your dependencies.
```

File: content/docs/user-guide/build/distributable-engine.md [hunk 6]
```diff
@@ -224,3 +224,21 @@ Engine developers will work on O3DE source code and create new distributable bui
...
+
+License attribution files can be generated during the project development process to properly attribute any code or packages that were imported (often called the NOTICES file). To scan project directories for licenses, project developers can run a script located in the engine's `scripts\license_scanner` subdirectory.
+
+In this example, with the engine, project, and `C:\o3de-source\3rdParty` as your 3rd Party packages folder, run the following script in the engine source folder:
```
Comment (willihay) [inline content/docs/user-guide/build/distributable-engine.md:232]: For consistency with other parts of the website, let's use a different name for the packages folder.
```suggestion
In this example, with the engine, project, and `C:\o3de-source\3rdParty` as your downloadable packages folder, run the following script in the engine source folder:
```

File: content/docs/user-guide/build/packages.md [hunk 7]
```diff
@@ -209,3 +209,8 @@ CMake functions used in this example:
 * `ly_pip_install_local_package_editable`: See the definition in `cmake/LYPython.cmake`.
 
 {{< /note >}}
+
+## License file generation
+License attribution files can be generated during the project development process to properly attribute any 3p packages that were imported (often called the NOTICES file). To scan 3p package directories for licenses, you can run a script located in the engine's scripts\license_scanner subdirectory. This script will look for the `PackageInfo.json` file in order to create a summary file of all package licenses with SPDX tags for easy reference.
```
Comment (willihay) [inline content/docs/user-guide/build/packages.md:214]: As noted previously, for consistency, use "downloadable" rather than "third-party" packages.
Also, since the tool is scanning both the project and package directories, I'd be clear about this, too.
```suggestion
License attribution files (often called the NOTICES file) can be generated during the project development process to properly attribute any code or downloadable packages that were imported. To scan project and package directories for licenses, you can run a script located in the engine's `scripts\license_scanner` subdirectory. This script will look for the `PackageInfo.json` file in order to create a summary file of all package licenses with SPDX tags for easy reference. 
```

File: content/docs/user-guide/packaging/windows-release-builds.md [hunk 8]
```diff
@@ -260,6 +260,10 @@ Follow the instructions in [Bundling Assets for a Project Game Release Layout](a
 
 You may notice that both your bundled assets are smaller than the default `engine.pak` that was created automatically when you created a project game release layout earlier. 
 
+## License file generation
+License attribution files can be generated during the project development process to properly attribute any code or packages that were imported (often called the NOTICES file). To scan project directories for licenses, you can run a script located in the engine's scripts\license_scanner subdirectory.
```
Comment (willihay) [inline content/docs/user-guide/packaging/windows-release-builds.md:264]: Updating to match the changes in the other version of this paragraph.
```suggestion
License attribution files (often called the NOTICES file) can be generated during the project development process to properly attribute any code or downloadable packages that were imported. To scan project and package directories for licenses, you can run a script located in the engine's `scripts\license_scanner` subdirectory. This script will look for the `PackageInfo.json` file in order to create a summary file of all package licenses with SPDX tags for easy reference.
```

File: content/docs/user-guide/build/distributable-engine.md [hunk 9]
```diff
@@ -224,3 +224,21 @@ Engine developers will work on O3DE source code and create new distributable bui
 Project developers will work on project-specific source code or use O3DE creative tools to add project content. To enable their workflows, project developers are required to _only_ pull a project from its hosted location.
 
 Whenever working on their project, project developers should launch the tools that were checked in as the distibutable engine build. These tools are located in the project's `install\bin\Windows\profile\Default` subdirectory.
+
+### License file generation
+
+License attribution files can be generated during the project development process to properly attribute any code or packages that were imported (often called the NOTICES file). To scan project directories for licenses, project developers can run a script located in the engine's `scripts\license_scanner` subdirectory.
```
Comment (willihay) [inline content/docs/user-guide/build/distributable-engine.md:230]: Moving the explanation next to the phrase it's defining.
```suggestion
License attribution files (often called the NOTICES file) can be generated during the project development process to properly attribute any code or packages that were imported. To scan project directories for licenses, project developers can run a script located in the engine's `scripts\license_scanner` subdirectory.
```

**Review by willihay — CHANGES_REQUESTED** [review]

File: content/docs/user-guide/build/packages.md (hunk 7 above)
Comment (amzn-changml) [reply content/docs/user-guide/build/packages.md]: Seems reasonable, the project doesn't often contain a `PackageInfo.json` file (generally created for packages only)g, but it can happen.

**Review by willihay — APPROVED** [review]

**Review by brianherrera — APPROVED** [review]
